process/io/plot_proc.py

```diff
@@ -4384,6 +4384,154 @@ def plot_radprofile(prof, mfile_data, scan, impp, demo_ranges) -> float:
+        "radius_plasma_pedestal_density_norm"
+    ].get_scan(scan)
+    radius_plasma_pedestal_temp_norm = mfile_data.data[
+        "radius_plasma_pedestal_temp_norm"
+    ].get_scan(scan)
+
+    if i_plasma_pedestal == 0:
+        # Intialise the radius
+        rho = np.linspace(0, 1.0, n_plasma_profile_elements)
+
+        # The density profile
+        ne = nd_plasma_electron_on_axis * (1 - rho**2) ** alphan
+
+        # The temperature profile
+        te = temp_plasma_electron_on_axis_kev * (1 - rho**2) ** alphat
+
+    if i_plasma_pedestal == 1:
+        rho = np.linspace(0, 1, n_plasma_profile_elements)
+
+        # The density and temperature profile
+        ne = np.zeros_like(rho)
+        te = np.zeros_like(rho)
+        for q in range(rho.shape[0]):
+            if rho[q] <= radius_plasma_pedestal_density_norm:
+                ne[q] = (
+                    nd_plasma_pedestal_electron
+                    + (ne0 - nd_plasma_pedestal_electron)
+                    * (1 - rho[q] ** 2 / radius_plasma_pedestal_density_norm**2)
+                    ** alphan
+                )
+            else:
+                ne[q] = nd_plasma_separatrix_electron + (
+                    nd_plasma_pedestal_electron - nd_plasma_separatrix_electron
+                ) * (1 - rho[q]) / (
+                    1 - min(0.9999, radius_plasma_pedestal_density_norm)
+                )
+
+            if rho[q] <= radius_plasma_pedestal_temp_norm:
+                te[q] = (
+                    temp_plasma_pedestal_kev
+                    + (te0 - temp_plasma_pedestal_kev)
+                    * (1 - (rho[q] / radius_plasma_pedestal_temp_norm) ** tbeta)
+                    ** alphat
+                )
+            else:
+                te[q] = temp_plasma_separatrix_kev + (
+                    temp_plasma_pedestal_kev - temp_plasma_separatrix_kev
+                ) * (1 - rho[q]) / (1 - min(0.9999, radius_plasma_pedestal_temp_norm))
+
+    # Intailise the radiation profile arrays
+    pimpden = np.zeros([imp_data.shape[0], te.shape[0]])
+    lz = np.zeros([imp_data.shape[0], te.shape[0]])
+    prad = np.zeros(te.shape[0])
+
+    # Intailise the impurity radiation profile
+    for k in range(te.shape[0]):
+        for i in range(imp_data.shape[0]):
+            if te[k] <= imp_data[i][0][0]:
+                lz[i][k] = imp_data[i][0][1]
+            elif te[k] >= imp_data[i][imp_data.shape[1] - 1][0]:
+                lz[i][k] = imp_data[i][imp_data.shape[1] - 1][1]
+            else:
+                # Use np.interp for log-log interpolation
+                log_te_data = np.log([row[0] for row in imp_data[i]])
+                log_lz_data = np.log([row[1] for row in imp_data[i]])
+                lz[i][k] = np.exp(np.interp(np.log(te[k]), log_te_data, log_lz_data))
+            pimpden[i][k] = imp_frac[i] * ne[k] * ne[k] * lz[i][k]
+
+        for l in range(imp_data.shape[0]):  # noqa: E741
+            prad[k] = prad[k] + pimpden[l][k] * 1.0e-6
```

Comment:
Pretty confusing with `i, k, l`. I don't know what the axes of these arrays are that are being looped over.